File: tests/benchmarks/resolution/fixtures/julia/service.jl
module Service

using ..Repository
using ..Validators

function create_user(repo, name, email)
    valid_name = Validators.validate_name(name)
    valid_email = Validators.validate_email(email)
    if valid_name && valid_email
        Repository.save(repo, name, email)
    else
        error("validation failed")
    end
end

function get_user(repo, id)
    Repository.find_by_id(repo, id)
end

function remove_user(repo, id)
    Repository.delete(repo, id)
end

function summary(repo)
    cnt = Repository.count(repo)
    format_summary(cnt)
end

function format_summary(cnt)
    "repository contains $cnt users"
end

end # module Service

File: tests/benchmarks/resolution/fixtures/erlang/service.erl
-module(service).
-export([create_user/4, get_user/2, remove_user/2]).

create_user(Id, Name, Email, Table) ->
    case validators:validate_user_input(Name, Email) of
        ok ->
            User = #{id => Id, name => Name, email => Email},
            repository:save(User, Table);
        Error ->
            Error
    end.

get_user(Id, Table) ->
    repository:find_by_id(Id, Table).

remove_user(Id, Table) ->
    case repository:find_by_id(Id, Table) of
        {ok, _} ->
            repository:delete(Id, Table);
        Error ->
            Error
    end.
